File: src/defaults/macd.js
function* macd(C) {
    // DIFF : EMA(CLOSE,SHORT) - EMA(CLOSE,LONG);//短周期与长周期的收盘价的指数平滑移动平均值做差。
    // DEA  : EMA(DIFF,M);//DIFF的M个周期指数平滑移动平均
    // 2*(DIFF-DEA),COLORSTICK;//DIFF减DEA的2倍画柱状线
    C.DEFINE({
        type: "SUB",
        cname: "MACD",
        state: "KLINE",
        yaxis: [
            {id: 0, mid: 0, format: "NUMBER2"},
        ]
    });
    //参数
    let vshort = C.PARAM(20, "SHORT", {MIN: 5, STEP: 5});
    let vlong = C.PARAM(35, "LONG", {MIN: 5, STEP: 5});
    let vm = C.PARAM(10, "M", {MIN: 5, STEP: 5});
    //输入
    let sclose = C.SERIAL("CLOSE");
    //输出
    let diff = C.OUTS("LINE", "diff", {color: RED});
    let dea = C.OUTS("LINE", "dea", {color: BLUE, width: 2});
    let bar = C.OUTS("BAR", "bar", {color: RED});
    //临时序列
    let eshort = [];
    let elong = [];
    //计算
    while(true) {
        let i = yield;
        eshort[i] = EMA(i, sclose, vshort, eshort);
        elong[i] = EMA(i, sclose, vlong, elong);
        diff[i] = eshort[i] - elong[i];
        dea[i] = EMA(i, diff, vm, dea);
        bar[i] = 2 * (diff[i] - dea[i]);
    }
}

File: script/cases/macd.py
#!/usr/bin/env python
#coding=utf-8

case = {
    "id": "MACD",
    "cname": "MACD",
    "type": "SUB",
    "src": """
//该模型仅仅用来示范如何根据指标编写简单的模型
//用户需要根据自己交易经验，进行修改后再实际应用!!!
// //后为文字说明，编写模型时不用写出
DIFF : EMA(CLOSE,SHORT) - EMA(CLOSE,LONG);//短周期与长周期的收盘价的指数平滑移动平均值做差。
DEA  : EMA(DIFF,M);//DIFF的M个周期指数平滑移动平均
2*(DIFF-DEA),COLORSTICK;//DIFF减DEA的2倍画柱状线

""",
    "params": [
        ("SHORT", 1, 100, 12),
        ("LONG", 1, 100, 26),
        ("M", 1, 100, 9),
    ],
    "expected": """
    """,
}
